### PR TITLE
[Feature][Connector V2] expose configurable options in Cassandra

### PR DESCRIPTION
**Pull request by cason0126**
#3663 
expose configurable options in Cassandra

### CODE REVIEW

File: seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/config/CassandraParameters.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        }
+        if (config.hasPath(CassandraConfig.BATCH_TYPE.key())) {
+            this.batchType = DefaultBatchType.valueOf(config.getString(CassandraConfig.BATCH_TYPE.key()));
+        } else {
+            this.batchType = DefaultBatchType.valueOf(CassandraConfig.BATCH_TYPE.defaultValue());
+        }
+        if (config.hasPath(CassandraConfig.ASYNC_WRITE.key())) {
+            this.asyncWrite = config.getBoolean(CassandraConfig.ASYNC_WRITE.key());
+        } else {
+            this.asyncWrite = true;
```
Comment (TaoZex) [inline seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/config/CassandraParameters.java:94]: The default value is assigned when the variable is declared. So we can get rid of the else branch.

File: seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/config/CassandraParameters.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        }
+        if (config.hasPath(CassandraConfig.BATCH_SIZE.key())) {
+            this.batchSize = config.getInt(CassandraConfig.BATCH_SIZE.key());
+        } else {
+            this.batchSize = CassandraConfig.BATCH_SIZE.defaultValue();
+        }
+        if (config.hasPath(CassandraConfig.BATCH_TYPE.key())) {
+            this.batchType = DefaultBatchType.valueOf(config.getString(CassandraConfig.BATCH_TYPE.key()));
+        } else {
+            this.batchType = DefaultBatchType.valueOf(CassandraConfig.BATCH_TYPE.defaultValue());
```
Comment (TaoZex) [inline seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/config/CassandraParameters.java:89]: Same as above.

Comment (cason0126) [reply seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/config/CassandraParameters.java:89]: ok。

Comment (cason0126) [reply seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/config/CassandraParameters.java:89]: But I was refered  to the writing of Redis.
Is this necessary at this stage?

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/config/CassandraParameters.java:89]: This is only suggested, not required.

**Comment by cason0126** [comment]
should i fix this intergration-test？ and how？ is there anything to refer to

**Comment by TaoZex** [comment]
> should i fix this intergration-test？ and how？ is there anything to refer to

image
It seems to be the problem of the code. Please locate and modify your code according to the error reported by CI.

**Comment by EricJoy2048** [comment]
Please fix CI

**Comment by cason0126** [comment]
> Please fix CI

I will fix them soon！

**Comment by cason0126** [comment]
is there anything to fix?
@EricJoy2048 @TaoZex

File: seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/sink/CassandraSinkFactory.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public String factoryIdentifier() {
+        return "Cassandra";
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+                .required(CassandraConfig.HOST, CassandraConfig.KEYSPACE, CassandraConfig.TABLE)
+                .optional(
+                        CassandraConfig.USERNAME, CassandraConfig.PASSWORD, CassandraConfig.DATACENTER,
```
Comment (FWLamb) [inline seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/sink/CassandraSinkFactory.java]: `USERNAME` and `PASSWORD` are bound.

File: seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/source/CassandraSourceFactory.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public String factoryIdentifier() {
+        return "Cassandra";
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+                .required(CassandraConfig.HOST, CassandraConfig.KEYSPACE, CassandraConfig.CQL)
+                .optional(CassandraConfig.USERNAME, CassandraConfig.PASSWORD, CassandraConfig.DATACENTER, CassandraConfig.CONSISTENCY_LEVEL)
```
Comment (FWLamb) [inline seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/source/CassandraSourceFactory.java]: `USERNAME` and `PASSWORD` are bound.

**Review by FWLamb — DISMISSED** [review]

**Comment by TyrantLucifer** [comment]
@FWLamb @EricJoy2048 PTAL.

**Review by hailin0 — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**